cpukit/score/src/threadprioritynode.c

```diff
@@ -0,0 +1,184 @@
+
+  /* If the_thread is at the start of the holder's Inherited_priorites list,
+   * then may need to update the holder's current priority. */
+  if ( _Chain_Is_first( &thread_pn->Node ) ) {
+    return true;
+  }
+  return false;
+}
+
+/**
+ * This routine sets the thread's priority_node->current_priority to the
+ * minimum of the real_priority and the first node of Inherited_priorities,
+ * if any.
+ */
+void _Thread_Evaluate_priority(
+  Thread_Control *the_thread
+)
+{
+  Chain_Node *first;
+  Thread_Priority_node *first_pn, *priority_node;
+  Priority_Control current_priority, min_priority;
+
+  /* Start with the_thread, and iterate through the holder threads that
+   * the_thread is waiting on. If any thread's priority does not change,
+   * then stop iterating. */
+  while ( the_thread != NULL ) {
+    priority_node = &the_thread->Priority_node;
+    current_priority = priority_node->current_priority;
+    min_priority = priority_node->real_priority;
+    first = _Chain_First( &priority_node->Inherited_priorities );
+    if ( !_Chain_Is_tail( &priority_node->Inherited_priorities, first ) ) {
+      first_pn = (Thread_Priority_node *)first;
+      if ( min_priority > first_pn->current_priority ) {
+        min_priority = first_pn->current_priority;
+      }
+    }
+    if ( current_priority < min_priority ) {
+        _Thread_Restore_priority(
+                the_thread
+                );
+    } else if ( current_priority > min_priority ) {
+        _Thread_Raise_priority(
+                the_thread,
+                min_priority
+                );
+    } else {
+      break;
+    }
+    if ( the_thread->Priority_node.waiting_to_hold != NULL ) {
+      the_thread = the_thread->Priority_node.waiting_to_hold->holder;
+    } else {
+      the_thread = NULL;
+    }
+  }
+}
+
+/**
+ * This routine removes a thread's Priority_node from an
+ * Inherited_priorities list, updates the mutex holder's
+ * current_priority if needed, and clears the mutex field.
+ *
+ * Returns the mutex that the_thread was waiting on.
+ */
+CORE_mutex_Control* _Thread_Dequeue_priority_node(
+  Thread_Priority_node *thread_pn
+)
+{
+  CORE_mutex_Control *the_mutex = thread_pn->waiting_to_hold;
+
+  if ( the_mutex == NULL ) {
+    return NULL;
+  }
+
+  _Chain_Extract_unprotected( &thread_pn->Node );
+  thread_pn->waiting_to_hold = NULL;
+
+  return the_mutex;
+}
+
+void _Thread_Requeue_priority_node(
+  Thread_Control *the_thread
+)
+{
+  ISR_Level       level;
+  CORE_mutex_Control *mutex;
+  bool mutex_owner_needs_update;
+
+  _ISR_Disable( level );
+  /* TODO: refactor? */
+  /* We are merely removing the priority node from 
+   * OwnerThread.Inherited_priorities
+   * and readding the the_thread->Priority_node back
+   */
+  mutex = _Thread_Dequeue_priority_node( &the_thread->Priority_node );
+  if ( mutex != NULL ) {
+    mutex_owner_needs_update = _Thread_Enqueue_priority_node( the_thread, mutex
+            );
```

Comment:
line wrapping

Reply:
these were due to my vim config.. fixing it now.